### ISSUE TITLE
UI scaling off

### ISSUE THREAD

**Issue by whtv** [CDrummond/cantata#1063]
I get wrongly scaled widgets in 2.1.0. Everything is huge, except the fonts. It looks about right when I set QT_AUTO_SCREEN_SCALE_FACTOR=0, but then some icons are missing and cantata crashes when I try to open preferences (which might not be related after all).
I've been using 2.0.1, from the Arch repos, with no such problems. All my packages are up to date (the Arch cantata package hasn't been updated to 2.1.0 yet, so I compiled from source).
Sorry if this is the wrong place to look for support. I don't use any Qt applications aside from cantata (love it!), so I have no experience with problems like this one. I did google, though :)

**Comment by CDrummond** [comment]
I dont see this. What Qt version? What is your screen DPI? Are you on a high DPI screen? Cantata looks fine on my full-hd laptop, and on a non-hd screen via VirtualBox.

**Comment by CDrummond** [comment]
If you edit `support/utils.cpp` and replace line number 842 from:
```
    static double scaleFactor=-1.0;
```

to
```
    static double scaleFactor=1.0;
```

Does this fix the issue? If so, then this is Cantata's fake high-dpi scaling causing issues.

**Comment by CDrummond** [comment]
Can you checkout git/master - I've removed Cantata's dpi stuff, so this should help. Is this better?

**Comment by whtv** [comment]
Sadly, both made no difference. I even tried removing the AA_EnableHighDpiScaling in gui/main.cpp (not that I know what I'm doing), to no avail. I also did some better googling this time and it seems like this is a problem in Qt that has recently re-occurred (e. g. https://bugzilla.redhat.com/show_bug.cgi?id=1381828).

To answer your questions, still: I'm on a full-hd laptop too, DPI as reported by X is 96, but the real DPI is around 176. I guess Qt regards that as a high DPI screen. Qt version is 5.9.1.

Thanks for the help!

**Comment by CDrummond** [comment]
I know you don't use any other Qt applications - but please install VLC, or another Qt5 app (such as one mentioned on your linked bug report), just to check if its a Qt5 issue.

**Comment by whtv** [comment]
Alright, I tried VLC, Calibre and Spyder. VLC and Spyder look fine, Calibre looks horrible, though I'm not sure what is actually off there and what is just ugly interface design.
I made some screenshots, also of Cantata 2.0.1 and 2.1.0 (notice the missing icons in 2.1.0):

![calibre](https://user-images.githubusercontent.com/28059402/29360191-0e88a1fc-8282-11e7-92e3-9495840db7fd.png)
![cantata-2 0 1](https://user-images.githubusercontent.com/28059402/29360194-0e9a0816-8282-11e7-857a-41529839847d.png)
![cantata-2 1 0](https://user-images.githubusercontent.com/28059402/29360192-0e96ea5a-8282-11e7-95d0-53bf5cb949ad.png)
![spyder](https://user-images.githubusercontent.com/28059402/29360450-3a0d43c2-8283-11e7-9823-773188010297.png)
![vlc](https://user-images.githubusercontent.com/28059402/29360193-0e983b3a-8282-11e7-89ed-81b0b78bb83e.png)

**Comment by CDrummond** [comment]
Have you installed Cantata 2.1.0 ? The missing icons are all from font-awesome. Cantata installs a copy of this into /usr/share/cantata - and loads if from there. If Cantata is not installed, then it will not be able to load this font file.

Can you also try Cantata with a different widget style? e.g. From a commandline:
```
cantata --style fusion
```

**Comment by whtv** [comment]
Okay, this is funny. It's actually working now after a reboot. What the hell? And after a make install the icons are fixed, too, all good now! The top bar with the player controls is still bigger than in 2.0.1 but that's fine.
That Calibre app still looks like before. So apparently your changes to the code did work after all but somehow didn't take effect on my end until after a reboot?
Anyway, thanks again for all the effort!

**Comment by whtv** [comment]
Wait. I forgot that I had removed the Qt::AA_EnableHighDpiScaling flag. Turns out that made the difference, I just double-checked. Not sure why it wouldn't take effect before a restart, but I can confirm now that setting this flag leads to the mis-scaling.

**Comment by CDrummond** [comment]
Hmmm...  Qt::AA_EnableHighDpiScaling was added by another user. Not sure if its required or not!

As to needing a restart - are you sure you don't have Cantata in the system try? ie. when you close it its not really stopped.

**Comment by whtv** [comment]
Yeah, I have the tray icon disabled in the preferences. Also I switched back and forth from 2.0.1 to 2.1.0 as you can see in the screenshots, so it wasn't that.
About the flag, I don't know anything about this obviously, but a quick look at the VLC code shows that the'yre using the flag too and VLC doesn't have the scaling issue for me: http://git.videolan.org/?p=vlc.git;a=blob;f=modules/gui/qt/qt.cpp;h=f521d04171e8fdee95fa03dd936bd0ce56efb9e9;hb=HEAD

**Comment by whtv** [comment]
Note that the flag can be overridden anyway by setting QT_AUTO_SCREEN_SCALE_FACTOR=0, so you might just leave it set as it is easy enough to workaround.

**Comment by CDrummond** [comment]
So, are you saying with "QT_AUTO_SCREEN_SCALE_FACTOR=0" that the issue is now resolved? If that's the case, then it surely must be a Qt issue?

**Comment by whtv** [comment]
That's right. Setting that environment variable has the same effect as removing the flag from the code. It turns the Qt5 auto-scaling off (it holds a boolean value).

**Comment by CDrummond** [comment]
Can you checkout git/master? I've changed the code so that the icon sizes are hard-coded. I'd be interested to see if you still have the issue with the play/pause/next/prev buttons. Probably will, but...

**Comment by whtv** [comment]
The player control buttons buttons have the correct size, before and after your commit (I couldn't see a difference). What is in fact different from 2.0.1 is that the top bar/widget itself is wider, and the info and menu button are larger. See attached screenshot and compare with 2.0.1 from above (the first of the two Cantata screenshots). I can live with the wider top bar. I consider this issue resolved for me: Qt auto-scaling doesn't seem to work right, so I switch that off system-wide by setting above mentioned evironment variable, and forget about it.
![2017-08-18-111448_1920x1080_scrot](https://user-images.githubusercontent.com/28059402/29452970-619fe468-8408-11e7-94cd-ee94f7db81dc.png)

**Comment by CDrummond** [comment]
Top bar being wider - I assume you mean taller? Not sure why tours is so tall in relation to the play controls size. Can you try editing gui/mainwindow.cpp and change line 230 from
```
toolbar->setMinimumHeight(fontMetrics().height()*4);
```
to
```
toolbar->setFixedHeight(fontMetrics().height()*4);
```

Does that help?

---

As to the German translation, feel free to either create a pull request, or just email me the updated file. My email address is in the AUTHORS file.

**Comment by whtv** [comment]
That didn't help. I then changed the factor to 3 and now it is slightly smaller than in 2.0.1. Looks good. Again, no apparent difference between setMinimumHeight and setFixedHeight.

**Comment by CDrummond** [comment]
OK, I've set it to a fixed height of 3.5 time font size.
